core/src/header.rs

```diff
@@ -230,6 +230,10 @@ impl<I, P> DataElement<I, P> {
         &self.value
     }
 
+    pub fn value_mut(&mut self) -> &mut Value<I, P> {
```

Comment:
This can make it too easy for the value to mismatch with the length described in the header. I was hoping not to introduce more gaps in which mismatches can be introduced (although the writer is safe from this and the case for text encoding conversion is known), which is why the `update_value` method was made instead. Is there something that you cannot perform with this 
method?

Reply:
IIRC I added it since it was needed in `value_at_mut`, but that one is being nuked anyway so this can go away as well